samtranslator/swagger/swagger.py

```diff
@@ -788,6 +789,32 @@ def add_models(self, models):
 
             self.definitions[model_name.lower()] = schema
 
+    def add_resource_policy(self, resource_policy):
+        """
+        Add resource policy definition to Swagger.
+
+        :param dict resource_policy: Dictionary of resource_policy statements which gets translated
+        :return:
+        """
+        if resource_policy is None:
+            return
+
+        custom_statements = resource_policy.get('CustomStatements')
+        if custom_statements is not None:
+            self.resource_policy['Version'] = '2012-10-17'
+            if self.resource_policy.get('Statement') is None:
+                self.resource_policy['Statement'] = custom_statements
+            else:
+                statement = self.resource_policy['Statement']
+                if type(statement) is list:
+                    statement.extend(custom_statements)
```

Comment:
Do we specifically validate that what's in `CustomStatements` ahead of this code? If not, this could blow up with a 5xx. Should we support them just adding a single statement that's not a list (is a dict)?

Reply:
I don't think my question has been addressed in the latest version of this PR. Are we validating custom_statements is a list? If not, this can blow up if custom_statements is not a list, e.g., dictionary, string.

Reply:
True. This only works if `custom_statements` is an iterable. If it's anything else, this code will fail.